### DISCUSSION TITLE
Couple questions regarding events

### DISCUSSION THREAD

**Discussion by kellynichols-adsk** [bbc/sqs-consumer#439] [Help] [answered]
### Summary

Assuming no timeout or batch processing, will either the `message_processed` or `processing_error` event be emitted for all invocations of the message handler? I'm managing some state that I want to make sure to clean up after a message is handled (successfully or not).

Also, for what errors would the `error` event be emitted with a `message` arg?

Thanks!

**Comment by nicholasgriffintn** [comment, marked as answer]
Events are documented here:

https://bbc.github.io/sqs-consumer/interfaces/Events.html

Hopefully that goes some way to explain. In terms of error, the message is only included if it's available:

https://github.com/bbc/sqs-consumer/blob/298fe5bfd29c548f99618ea26b8e37d0a502e0fe/src/consumer.ts#L173
